src/task_types/burn_in_tests.rs

```diff
@@ -220,41 +223,75 @@ impl BurnInServiceImpl {
             Some(origin_suite),
         )
     }
+
+    /// Determines whether to use large distros based on the list of tasks that explicitly specify distros;
+    /// assuming those are the tasks that need to use large distros.
+    ///
+    /// # Arguments
+    ///
+    /// * `variant_task_refs` - Vec of of task refs on a build variant definition with the distro information.
+    /// * `discovered_tasks` - Vec of tasks determined to be run by burn_in.
+    ///
+    /// # Returns
+    ///
+    /// a boolean of whether a large distro should be used for burn_in.
+    fn should_use_large_distro(
+        &self,
+        variant_task_refs: &Vec<TaskRef>,
+        discovered_tasks: &Vec<DiscoveredTask>,
+    ) -> bool {
+        let mut large_distro_tasks = HashSet::new();
+
+        for task_ref in variant_task_refs {
+            if task_ref.distros.is_some() {
+                large_distro_tasks.insert(task_ref.name.clone());
+            }
+        }
+
+        for discovered_task in discovered_tasks {
+            if large_distro_tasks.contains(&discovered_task.task_name) {
+                return true;
```

Comment:
So with this if any of the discovered tasks will have `use_large_distro: true` option, all burn_in sub tasks will run on a large distro, even if some of them are not running on large distro in a normal run.

I think if we would just default `use_large_distro` to `true` for all burn_in_tests, it will be very much the same and much simpler. Just as a quick fix to reduce complaints about burn_in_tests failures due to average time comparison with historic data.

To make those average test time checks in burn_in_tests more valuable I think we should move this [use_large_distro](https://github.com/mongodb/mongo-task-generator/blob/master/src/task_types/resmoke_tasks.rs#L231) from display_task to sub_suite level and set it in accordance to the original task definition.

Reply:
I agree with Misha, I think we should this to be tracked at the sub_suite level.

I think a lot of the value from `burn_in` comes from being able to see at a glance if your changes have introduced an easily observable change in runtime. So, I think we should try to replicate the environment on a sub task by sub task basis if possible so that we're not accidentally hiding runtime increases by running things on a bigger, faster machine.

Reply:
Yeah valid points. Let me update the sub_suite part to accommodate task-level properties like distro sizes